src/Hosting/TestHost/src/ClientHandler.cs

```diff
@@ -65,7 +66,39 @@ internal ClientHandler(PathString pathBase, ApplicationWrapper application)
             var contextBuilder = new HttpContextBuilder(_application, AllowSynchronousIO, PreserveExecutionContext);
 
             var requestContent = request.Content ?? new StreamContent(Stream.Null);
-            var body = await requestContent.ReadAsStreamAsync();
+
+            var requestPipe = new Pipe();
+
+            var copyRequestTask = Task.Run(async () =>
```

Comment:
One behavior change here is that this initiates client side activity before the app starts reading the request body. I'm not sure what the consequences of that will be in people's tests. In a normal app that would break 100-continue, but in a test app it might be OK because the client doesn't have any automatic behaviors like redirects or auth.

Reply:
It might be worth fixing this by passing a func to the stream and calling it the first time ReadAsync is called.

Reply:
> One behavior change here is that this initiates client side activity before the app starts reading the request body.

This isn't a change from how TestServer currently works. If you're passing in `StreamContent` then that stream is read from directly. Any other `HttpContent` implementation was already being eagerly read before the server stated via `ReadAsStreamAsync`.

Reply:
> If you're passing in StreamContent then that stream is read from directly.

But the timing is different. It now starts reading that stream immediately, not when the app starts reading it.

> Any other HttpContent implementation

... that didn't implement CreateContentReadStream.

I'm not strictly worried about this part of the change, I just wanted to call out that the timing change and it may affect tests. All of our tests passed so that implies the effect is limited.

Reply:
Yes you're right. Ok!